activerecord/lib/active_record/transactions.rb

```diff
@@ -358,7 +358,7 @@ def with_transaction_returning_status
       connection = self.class.connection
       ensure_finalize = !connection.transaction_open?
 
-      connection.transaction do
+      connection.transaction(mode: :immediate) do
```

Comment:
Don't want to have a SQLite3 specific mode concern leak all the way into the other adapters and the root transaction setup. There's gotta be a better way with less collateral damage.

Reply:
When I had focused on just the fixtures, I introduced `use_deferred_transaction_mode!`, `use_immediate_transaction_mode!`, and `use_default_transaction_mode!` methods to the SQLite3 adapter and called them before calling `#transaction`. This original implementation was to avoid precisely this leak, tho it traded that with "spooky action as a distance". When I pivoted to this approach, I thought that some users are likely to want control over the transaction mode when starting a transaction manually. From this use-case point of view, I thought that the "spooky action at a distance" approach of:
```ruby
connection.use_immediate_transaction_mode!
connection.transaction do
end
```
felt off.

I am open to going in that direction if you think that the conceptual leak of the reality that SQLite has transaction modes and other DBMSes don't is a larger issue than the interface for end-user control of SQLite's transaction modes. 

I did make sure to ensure that none of the other adapters change their interface, only their method signature. A couple other possibilities:

1. implement a generic `options = {}` interface for the `transaction` method where currently only the SQLite adapter looks into the hash to see if a `:mode` was set
2. keep the `mode = nil` interface for the `transaction` method, but explicitly `raise` if that option is passed into the methods of the other adapters

Or, again, we could revert to the `connection.use_immediate_transaction_mode!` approach I was using earlier. Of course, I'm also open to any other suggestions.